### PR TITLE
fix: date-picker not displaying calendar widget in mobile devices

### PR DESCRIPTION
**Pull request by ravise5**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #<gh-issue-id>

Test URLs:
- Before: https://main--aem-boilerplate-forms--adobe-rnd.hlx.live/drafts/rae/test
- After: https://dpfix--aem-boilerplate-forms--adobe-rnd.hlx.live/drafts/rae/test

### CODE REVIEW

File: blocks/form/form.js [hunk 1]
```diff
@@ -285,6 +289,7 @@ function inputDecorator(field, element) {
       input.setAttribute('display-value', field.displayValue ?? '');
       input.type = 'text';
       input.value = field.displayValue ?? '';
+      input.addEventListener('touchstart', () => handleTouchStart(input, field)); // in mobile devices the input type needs to be toggled before focus
```
Comment (nit23uec) [inline blocks/form/form.js]: we should figure out a way to automate such mobile specific testing. Does playwright gives mobile emulation?

Comment (ravise5) [reply blocks/form/form.js]: Yes, playwright has this OOTB

Comment (ravise5) [reply blocks/form/form.js]: I can add a test case

Comment (vdua) [reply blocks/form/form.js]: no need to create a function for that. 
```
input.addEventListener('touchstart', () => input.type = field.type)
```

**Review by nit23uec — APPROVED** [review]

**Pull request merged**